integration-tests/smoke/automation_test.go

```diff
@@ -1105,31 +1142,82 @@ func setupAutomationTestDocker(
 
 	env.ParallelTransactions(true)
 	nodeClients := env.ClCluster.NodeAPIs()
-	workerNodes := nodeClients[1:]
 
-	linkToken, err := env.ContractDeployer.DeployLinkTokenContract()
-	require.NoError(t, err, "Error deploying LINK token")
+	a := automationv2.NewAutomationTestDocker(env.EVMClient, env.ContractDeployer, nodeClients)
+	a.MercuryCredentialName = "cred1"
+	a.RegistrySettings = registryConfig
+	a.RegistrarSettings = contracts.KeeperRegistrarSettings{
+		AutoApproveConfigType: uint8(2),
+		AutoApproveMaxAllowed: 1000,
+		MinLinkJuels:          big.NewInt(0),
+	}
+	a.PluginConfig = ocr2keepers30config.OffchainConfig{
+		TargetProbability:    "0.999",
+		TargetInRounds:       1,
+		PerformLockoutWindow: 3_600_000, // Intentionally set to be higher than in prod for testing purpose
+		GasLimitPerReport:    10_300_000,
+		GasOverheadPerUpkeep: 300_000,
+		MinConfirmations:     0,
+		MaxUpkeepBatchSize:   10,
+	}
+	a.PublicConfig = ocr3.PublicConfig{
+		DeltaProgress:                           10 * time.Second,
+		DeltaResend:                             15 * time.Second,
+		DeltaInitial:                            500 * time.Millisecond,
+		DeltaRound:                              1000 * time.Millisecond,
+		DeltaGrace:                              200 * time.Millisecond,
+		DeltaCertifiedCommitRequest:             300 * time.Millisecond,
+		DeltaStage:                              30 * time.Second,
+		RMax:                                    24,
+		MaxDurationQuery:                        20 * time.Millisecond,
+		MaxDurationObservation:                  20 * time.Millisecond,
+		MaxDurationShouldAcceptAttestedReport:   1200 * time.Millisecond,
+		MaxDurationShouldTransmitAcceptedReport: 20 * time.Millisecond,
+		F:                                       1,
+	}
+
+	a.SetupAutomationDeployment(t)
+	a.SetDockerEnv(env)
 
-	registry, registrar := actions.DeployAutoOCRRegistryAndRegistrar(
-		t,
-		registryVersion,
-		registryConfig,
-		linkToken,
-		env.ContractDeployer,
-		env.EVMClient,
-	)
+	if isMercuryV02 || isMercuryV03 {
+		var imposters []ctfTestEnv.KillgraveImposter
+		mercuryv03Mock200 := ctfTestEnv.KillgraveImposter{
+			Request: ctfTestEnv.KillgraveRequest{
+				Method:     http.MethodGet,
+				Endpoint:   "/api/v1/reports/bulk",
```

Comment:
This looks good to me! We dont really differentiate mercury v20 or v30, we set both endpoints, params, etc to ENV.